.codesandbox/templates/next/pages/index.tsx

```diff
@@ -1,60 +1,70 @@
 
 const IndexPage: NextPage = () => {
-  const ref = useRef<fabric.Canvas>(null);
-
-  const onLoad = useCallback(
-    (canvas: fabric.Canvas) => {
-      canvas.setDimensions({
-        width: window.innerWidth,
-        height: 500,
-      });
-      const textValue = 'fabric.js sandbox';
-      const text = new fabric.Textbox(textValue, {
-        originX: 'center',
-        top: 20,
-        textAlign: 'center',
-        styles: fabric.util.stylesFromArray(
-          [
-            {
-              style: {
-                fontWeight: 'bold',
-                fontSize: 64,
-              },
-              start: 0,
-              end: 9,
+  const create = useCallback((arg0: fabric.DOMManagerType) => {
+    const canvas = (window.canvas = new fabric.Canvas(arg0));
+    canvas.setDimensions({
+      width: window.innerWidth,
+      height: 500,
+    });
+    const textValue = 'fabric.js sandbox';
+    const text = new fabric.Textbox(textValue, {
+      originX: 'center',
+      top: 20,
+      textAlign: 'center',
+      styles: fabric.util.stylesFromArray(
+        [
+          {
+            style: {
+              fontWeight: 'bold',
+              fontSize: 64,
             },
-          ],
-          textValue
-        ),
-      });
-      canvas.add(text);
-      canvas.centerObjectH(text);
+            start: 0,
+            end: 9,
+          },
+        ],
+        textValue
+      ),
+    });
+    canvas.add(text);
+    canvas.centerObjectH(text);
 
-      const animate = (toState: number) => {
-        text.animate(
-          { scaleX: Math.max(toState, 0.1) * 2 },
-          {
-            onChange: () => canvas.renderAll(),
-            onComplete: () => animate(Number(!toState)),
-            duration: 1000,
-            easing: toState
-              ? fabric.util.ease.easeInOutQuad
-              : fabric.util.ease.easeInOutSine,
-          }
-        );
-      };
-      animate(1);
-    },
-    [ref]
-  );
+    const animate = (toState: number) => {
+      text.animate(
+        { scaleX: Math.max(toState, 0.1) * 2 },
+        {
+          onChange: () => canvas.renderAll(),
+          onComplete: () => animate(Number(!toState)),
+          duration: 1000,
+          easing: toState
+            ? fabric.util.ease.easeInOutQuad
+            : fabric.util.ease.easeInOutSine,
+        }
+      );
+    };
+    animate(1);
+
+    return canvas;
+  }, []);
 
   return (
-    <div className="position-relative">
-      <Canvas ref={ref} onLoad={onLoad} />
+    <div style={{ position: 'relative' }}>
+      <CanvasComponent create={create}>
+        <CanvasSlot name="main" />
+        <CanvasSlot name="pointers" />
+        <iframe
+          src="https://fabricjs.github.io/"
+          width="100%"
+          height="500px"
+          style={{ position: 'absolute', left: 0, top: 0, opacity: 0.8 }}
+        />
+        Hello World
+        {/* keep on top */}
+        <CanvasSlot name="top" />
+      </CanvasComponent>
```

Comment:
This looks really funny on chrome, opacity leaks from the iframe to the main context
<img width="1512" alt="Screenshot 2024-02-29 at 21 06 57" src="https://github.com/fabricjs/fabric.js/assets/34343793/8864f118-82c7-4c95-a340-a92d8f3e7301">

Reply:
same in safari